components/validator/guardian.go

```diff
@@ -232,11 +239,19 @@ func (g *Guardian) outputRootAtBlock(ctx context.Context, blockNumber uint64) (e
 	return output.OutputRoot, nil
 }
 
-func (g *Guardian) sendTransaction(tx *types.Transaction) {
-	g.txCandidatesChan <- txmgr.TxCandidate{
-		TxData:     tx.Data(),
-		To:         tx.To(),
-		GasLimit:   tx.Gas(),
-		AccessList: nil,
+func (g *Guardian) sendTransaction(ctx context.Context, tx *types.Transaction) error {
+	txResponse := g.cfg.TxManager.SubmitTransaction(ctx, txmgr.TxCandidate{
+		TxData:   tx.Data(),
+		To:       tx.To(),
+		GasLimit: tx.Gas(),
+	})
+	if txResponse.Err != nil {
+		g.log.Error("failed to send tx with response", "err", txResponse.Err)
+		return txResponse.Err
 	}
+
+	if txResponse.Receipt.Status != types.ReceiptStatusSuccessful {
+		return fmt.Errorf("failed to send tx with receipt status. status: %+v", txResponse.Receipt.Status)
+	}
```

Comment:
This case already returns error in [`txmgr.send()`](https://github.com/kroma-network/kroma/blob/perf/add-buffered-txmgr/utils/service/txmgr/txmgr.go#L260-L263), we don't need to check this case here again!